subsys/bluetooth/audio/ascs.c

```diff
@@ -1446,6 +1502,34 @@ static ssize_t ascs_config(struct bt_ascs *ascs, struct net_buf_simple *buf)
 	return buf->size;
 }
 
+int bt_audio_ep_get_info(struct bt_audio_ep *ep, struct bt_audio_ep_info *info)
+{
+	info->id = ep->status.id;
+	info->state = ep->status.state;
+	info->dir = ep->dir;
+
+	return 0;
+}
+
+void bt_audio_unicast_server_foreach_ase(struct bt_conn *conn,
+		bt_audio_ase_func_t func, void *user_data)
+{
+	struct bt_audio_ep_info info;
+	struct bt_ascs_ase *ase;
+	sys_snode_t *ase_node;
+	struct bt_ascs *ascs = ascs_get(conn);
+
+	SYS_SLIST_FOR_EACH_NODE(&ascs->ases, ase_node) {
+		ase = CONTAINER_OF(ase_node, struct bt_ascs_ase, node);
+
+		if (ase == NULL)
+			return;
+
+		bt_audio_ep_get_info(&ase->ep, &info);
```

Comment:
As I suggested in `audio.h`, we should just return the `ep` pointer, and let the caller decide if they want to get the `info`